tests/test_requests.py

```diff
@@ -530,7 +530,9 @@ def test_basicauth_encodes_byte_strings(self):
             # Connecting to an invalid port should raise a ConnectionError
             ('http://localhost:1', ConnectionError),
             # Inputing a URL that cannot be parsed should raise an InvalidURL error
-            ('http://fe80::5054:ff:fe5a:fc0', InvalidURL)
+            ('http://fe80::5054:ff:fe5a:fc0', InvalidURL),
+            # Inputing a URL that cannot be parsed by urllib3 should raise a ConnectionError, as it closes pool
+            ('http://1234567890123456789012345678901234567890123456789012345678901234.com', ConnectionError)
```

Comment:
This test seems to contradict the behaviour we seem to be changing. At least it seems to indicate that we're not allowing a 3rd party exception to be raised